core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -319,6 +402,1345 @@ public void testNamespaceWithDot() {
     Assert.assertFalse("Namespace should not exist", catalog.namespaceExists(withDot));
   }
 
+  @Test
+  public void testBasicCreateTable() {
+    C catalog = catalog();
+
+    TableIdentifier ident = TableIdentifier.of("ns", "table");
+
+    Assert.assertFalse("Table should not exist", catalog.tableExists(ident));
+
+    Table table = catalog.buildTable(ident, SCHEMA).create();
+    Assert.assertTrue("Table should exist", catalog.tableExists(ident));
+
+    // validate table settings
+    Assert.assertEquals("Table name should report its full name", catalog.name() + "." + ident, table.name());
+    Assert.assertEquals("Schema should match expected ID assignment",
+        TABLE_SCHEMA.asStruct(), table.schema().asStruct());
+    Assert.assertNotNull("Should have a location", table.location());
+    Assert.assertTrue("Should be unpartitioned", table.spec().isUnpartitioned());
+    Assert.assertTrue("Should be unsorted", table.sortOrder().isUnsorted());
+    Assert.assertNotNull("Should have table properties", table.properties());
+  }
```

Comment:
Do we need to drop the tables that are being created?

Reply:
No, the catalog is re-initialized every time and data on disk is in a temporary folder.